planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -346,29 +350,33 @@ bool DefaultPlanner::is_goal_valid(
   const auto polygon_footprint = convert_linear_ring_to_polygon(goal_footprint);
 
   double next_lane_length = 0.0;
+  bool is_goal_valid = false;
   // combine calculated route lanelets
   lanelet::ConstLanelet combined_prev_lanelet = combine_lanelets(path_lanelets);
 
   // check if goal footprint exceeds lane when the goal isn't in parking_lot
-  if (
-    param_.check_footprint_inside_lanes &&
-    !check_goal_footprint(
-      closest_lanelet, combined_prev_lanelet, polygon_footprint, next_lane_length) &&
-    !is_in_parking_lot(
-      lanelet::utils::query::getAllParkingLots(lanelet_map_ptr_),
-      lanelet::utils::conversion::toLaneletPoint(goal.position))) {
-    RCLCPP_WARN(logger, "Goal's footprint exceeds lane!");
-    return false;
-  }
+  for (const auto & gl_llt : goal_lanelets) {
+    if (
+      param_.check_footprint_inside_lanes &&
+      !check_goal_footprint(gl_llt, combined_prev_lanelet, polygon_footprint, next_lane_length) &&
+      !is_in_parking_lot(
+        lanelet::utils::query::getAllParkingLots(lanelet_map_ptr_),
+        lanelet::utils::conversion::toLaneletPoint(goal.position))) {
+      RCLCPP_WARN(logger, "Goal's footprint exceeds lane!");
+      is_goal_valid = false;
+    } else {
+      is_goal_valid = true;
+    }
 
-  if (is_in_lane(closest_lanelet, goal_lanelet_pt)) {
-    const auto lane_yaw = lanelet::utils::getLaneletAngle(closest_lanelet, goal.position);
-    const auto goal_yaw = tf2::getYaw(goal.orientation);
-    const auto angle_diff = tier4_autoware_utils::normalizeRadian(lane_yaw - goal_yaw);
+    if (is_in_lane(gl_llt, goal_lanelet_pt)) {
+      const auto lane_yaw = lanelet::utils::getLaneletAngle(gl_llt, goal.position);
+      const auto goal_yaw = tf2::getYaw(goal.orientation);
+      const auto angle_diff = tier4_autoware_utils::normalizeRadian(lane_yaw - goal_yaw);
 
-    const double th_angle = tier4_autoware_utils::deg2rad(param_.goal_angle_threshold_deg);
-    if (std::abs(angle_diff) < th_angle) {
-      return true;
+      const double th_angle = tier4_autoware_utils::deg2rad(param_.goal_angle_threshold_deg);
+      if (std::abs(angle_diff) < th_angle) {
+        return true;
+      }
```

Comment:
If goal's footprint exceeds lane but the process below may return true, is it ok?
It is like the footprint check is ignored.

Reply:
It is ok for us because we don't have any motivation to check footprint.

Reply:
@kosuke55 I'm asking just to be sure, is your suggestion to completely remove the footprint check instead of an edit that will eliminate the logical issue here?

Reply:
@kosuke55 friendly ping

Reply:
@beyzanurkaya 
Sorry for the delay.

I am not sure if we should remove the footprint check, but the results of it appear to be unused.
This PR appears to return `true` even if `is_goal_valid` is `false` if one point of goal_lanelet_pt is in the goal lane and the angle is threshold only.
In what scene would you use the result of the footprint check?
(i.e., in what scene `return is_goal_valid;`)

Reply:
Would the conditional branch look like this?

![image](https://github.com/autowarefoundation/autoware.universe/assets/39142679/5be93226-e360-473c-ac73-b118b7f1fd1b)

> If goal's footprint exceeds lane but the process below may return true, is it ok?
It is like the footprint check is ignored.

I mentioned about 2,3 pattern.

> In what scene would you use the result of the footprint check?
(i.e., in what scene return is_goal_valid;)

I think this scnene is 3 or 5. 
5 will ignore the angle check.